### PR TITLE
Example for updating a connection profile

### PR DESCRIPTION
**Pull request by bernhardkaindl**
Update a property of a connection profile, looked up by connection id

### CODE REVIEW

File: examples/async/update-connection-async.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+async def update_connection(args: Dict[str, Any]) -> Any:
+    """Update the settings for [key][entry] of the 1st matching connection"""
+
+    connection_tuple = await connection_by_id(args["connection_identifier"])
+    configuration_group, setting = args["connection_setting"]
+    if connection_tuple:
+        connection_settings_proxy, settings = connection_tuple
+        # For compatibility with old tools, NM adds and prefers them, delete:
+        del settings["ipv4"]["addresses"]  # -> Use ["ipv4"]["address-data"]
```
Comment (igo95862) [inline examples/async/update-connection-async.py:55]: I think `.pop()` method is better than `del`. `del` has some nasty side effects.

File: examples/async/example_utilities_async.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+DBusTuple = Tuple[str, Any]
+ConnectionSettingsTuple = Tuple[NetworkConnectionSettings, Any]
+
+
+def DBusDictionaryValue(dict: Dict[str, DBusTuple], key: str) -> Any:
+    "Return the value of a DBus Dictionary Tuple ([0] is a signature)"
+    return dict[key][1]
+
+
+async def connection_by_id(conn_id: str) -> Optional[ConnectionSettingsTuple]:
```
Comment (igo95862) [inline examples/async/example_utilities_async.py:21]: This file shows a deficiency of the binds. They need more helper methods. (the functions that are actually not part of D-Bus interface)

I will look in to adding more of them tomorrow that can remove the need for this file.

**Review by igo95862 — CHANGES_REQUESTED** [review]

File: examples/async/example_utilities_async.py (hunk 2 above)
Comment (bernhardkaindl) [reply examples/async/example_utilities_async.py]: Rebased and force-pushed on top of #8

**Comment by bernhardkaindl** [comment]
@igo95862 rebased to current master so you can merge this PR without needing another rebase.

To work of course, it will need some form of `get_connections_by_id()` like from PR #8.

**Pull request merged**